src/socketgateway.cc

```diff
@@ -8,6 +8,10 @@
 
 #include "logging.h"
 
+#ifndef MSG_NOSIGNAL
```

Comment:
Why do we need this defines?

Reply:
Used to not get a `SIGPIPE` after a broken pipe during `send()`. Note that we also incorrectly use it for `recv` but according to the man page it doesn't apply there. We'd probably want to use it for our secondary ipc implem as well...

It's not available on OSX: https://nwat.xyz/blog/2014/01/16/porting-msg_more-and-msg_nosigpipe-to-osx/.